File: MineInAbyss-API/src/main/java/com/derongan/minecraft/mineinabyss/Ascension/Effect/AscensionEffect.java
package com.derongan.minecraft.mineinabyss.Ascension.Effect;

import org.bukkit.entity.Player;

public interface AscensionEffect {
    void applyEffect(Player player, int ticks);
    boolean isDone();
    int getRemainingTicks();
    void setRemainingTicks(int remainingTicks);
    void cleanUp(Player player);
}

File: src/main/java/com/derongan/minecraft/mineinabyss/ascension/effect/AscensionEffect.kt
package com.derongan.minecraft.mineinabyss.ascension.effect

import com.mineinabyss.idofront.time.TimeSpan
import org.bukkit.entity.Player

interface AscensionEffect {
    fun applyEffect(player: Player, ticks: Int)
    val isDone: Boolean
    val duration: TimeSpan
    fun cleanUp(player: Player)

    fun clone(): AscensionEffect
}